src/phoenix/session/session.py

```diff
@@ -70,13 +71,19 @@ def __init__(
         reference_dataset: Optional[Dataset] = None,
         corpus_dataset: Optional[Dataset] = None,
         trace_dataset: Optional[TraceDataset] = None,
+        default_umap_parameters: Optional[Dict[str, int]] = None,
         host: str = HOST,
         port: int = PORT,
     ):
         self.primary_dataset = primary_dataset
         self.reference_dataset = reference_dataset
         self.corpus_dataset = corpus_dataset
         self.trace_dataset = trace_dataset
+        self.config = (
+            UMAPParameters(default_umap_parameters)
+            if default_umap_parameters is not None
+            else None,
```

Comment:
might be simpler to actually not make this optional but actually have the defaults explicit and the passed in values be a partial - this way you don't have to specify every parameter and this class can serve as the source of truth.

Reply:
I made sure that not all parameters are required to be passed but couldn't use partial.